dessia_common/workflow/blocks.py

```diff
@@ -774,6 +774,68 @@ def _to_script(self, _) -> ToScriptElement:
         script = f"ModelAttribute(attribute_name='{self.attribute_name}', {self.base_script()})"
         return ToScriptElement(declaration=script, imports=[self.full_classname])
 
+   
+class GetModelAttribute(Block):
+    """
+    Fetch attribute of given object during workflow execution.
+
+    :param attribute_type: AttributeType variable that contain the model and the name of the attribute to select.
+    :param name: Name of the block.
+    :param position: Position of the block in canvas.
+    """
+
+    def __init__(self, attribute_type: AttributeType[Type], name: str = '', position: Tuple[float, float] = None):
+        self.attribute_type = attribute_type
+        parameters = inspect.signature(self.attribute_type.class_).parameters
+        inputs = [TypedVariable(type_=self.attribute_type.class_, name='Model')]
+        type_ = GetModelAttribute.get_attributes_type(self.attribute_type.name, parameters)
+        if type_:
+            outputs = [TypedVariable(type_=type_, name='Model attribute')]  
+        else:
+            outputs=[Variable(name='Model attribute')]
+        Block.__init__(self, inputs, outputs, name=name, position=position)
+
+    def equivalent_hash(self):
+        """ Custom hash function. Related to 'equivalent' method. """
+        classname = self.attribute_type.class_.__name__
+        return len(classname) + 7 * len(self.attribute_type.name)
+
+    def equivalent(self, other):
+        """ Return whether the block is equivalent to the other given or not. """
+        classname = self.attribute_type.class_.__name__
+        other_classname = other.attribute_type.class_.__name__
+        same_model = classname == other_classname
+        same_method = self.attribute_type.name == other.attribute_type.name
+        return Block.equivalent(self, other) and same_model and same_method
+
+    def evaluate(self, values, **kwargs):
+        """ Get input object's deep attribute. """
+        return [get_in_object_from_path(values[self.inputs[0]], f'#/{self.attribute_type.name}')]
+
+    def _to_script(self, _) -> ToScriptElement:
+        """ Write block config into a chunk of script. """
+        script = f"GetModelAttribute(attribute_type=AttributeType(" \
+                 f"{self.attribute_type.class_.__name__}, name='{self.attribute_type.name}')" \
+                 f", {self.base_script()})"
+        imports = [full_classname(object_=self.attribute_type, compute_for='instance'),
+                   full_classname(object_=self.attribute_type.class_, compute_for='class'),
+                   self.full_classname]
+        return ToScriptElement(declaration=script, imports=imports)
+    
+    @classmethod
+    def get_attributes_type(cls, attribute_name: str, parameters: dict):
+        """ Get type of attribute name of class."""
+        type_parameter = parameters.get(attribute_name)
+        if not type_parameter:
+            return None
+        try:
+            type_parameter.annotation
```

Comment:
Can we check `if hasattr(parameter, "annotation"):` instead of trying to catch an AttributeError ?

Reply:
Yes, we can. Done